tutorial/source/gplvm.ipynb

```diff
@@ -315,63 +319,67 @@
+    "Now comes the most interesting part. We know that the observed data $y$ has latent structure: in particular different datapoints correspond to different cell stages. We would like our GPLVM to learn this structure in an unsupervised manner. In principle, if we do a good job of inference then we should be able to discover this structure---at least if we choose reasonable priors. First, we have to choose the dimension of our latent space $X$. We choose $dim(X)=2$, since we would like our model to disentangle 'capture time' ($1$, $2$, $4$, $8$, $16$, $32$, and $64$) from cell branching types (TE, ICM, PE, EPI). Next, when we set the mean of our prior over $X$, we set the first dimension to be equal to the observed capture time. This will help the GPLVM discover the structure we are interested in and will make it more likely that that structure will be axis-aligned in a way that is easier for us to interpret."
    ]
   },
   {
    "cell_type": "code",
-   "execution_count": 4,
+   "execution_count": 15,
    "metadata": {},
    "outputs": [],
    "source": [
     "captured_time = y.new_tensor([int(cell_name.split(\" \")[0]) for cell_name in df.index.values])\n",
-    "# we scale the time info into the interval [0, 1]\n",
-    "time = (captured_time.log2() - 1) / 5\n",
+    "# we scale the time into the interval [0, 1]\n",
+    "time = captured_time.log2() / 6\n",
     "\n",
-    "# set the initial value for X\n",
-    "X_init = torch.zeros(y.size(1), 2)  # shape: 437 x 2\n",
-    "X_init[:, 0] = time"
+    "# we setup the mean of our prior over X\n",
+    "X_prior_mean = torch.zeros(y.size(1), 2)  # shape: 437 x 2\n",
+    "X_prior_mean[:, 0] = time"
    ]
   },
   {
    "cell_type": "markdown",
    "metadata": {},
    "source": [
-    "We will use a sparse version of Gaussian process to make the inference faster. Remember that we also need to define $X$ as a `Parameter` to be able to set prior and guide for it."
+    "We will use a sparse version of Gaussian process inference to make training faster. Remember that we also need to define $X$ as a `Parameter` so that we can set a prior and guide (variational distribution) for it."
    ]
   },
   {
    "cell_type": "code",
-   "execution_count": 5,
-   "metadata": {},
+   "execution_count": 28,
+   "metadata": {
+    "collapsed": true
+   },
    "outputs": [],
    "source": [
     "kernel = gp.kernels.RBF(input_dim=2, lengthscale=torch.ones(2))\n",
     "\n",
-    "# we clone to avoid the change of value of X_init during the inference\n",
-    "X = Parameter(X_init.clone())\n",
+    "# we clone here so that we don't change our prior during the course of training\n",
```

Comment:
is this correct?

Reply:
yes, it is correct! It seems that I cloned `X_init` in the prior too, which is unnecessary.